worldedit-core/src/main/java/com/sk89q/worldedit/extension/factory/parser/DefaultBlockParser.java

```diff
@@ -272,21 +280,41 @@ private BaseBlock parseLogic(String input, ParserContext context) throws InputPa
 
         if (state == null) {
             String typeString;
-            String stateString = null;
+
             int stateStart = blockAndExtraData[0].indexOf('[');
-            if (stateStart == -1) {
+            int nbtStart = blockAndExtraData[0].indexOf('{');
+            int typeEnd = stateStart == -1 ? nbtStart : stateStart;
```

Comment:
Avoid second index if not needed? Maybe expand into an `if` statement to keep the name.
```suggestion
            int typeEnd = stateStart == -1 ? blockAndExtraData[0].indexOf('{') : stateStart;
```

Reply:
This is used later so it's not unneeded